Homework_4/hw4.py

```diff
@@ -0,0 +1,94 @@
+user = {
+    'PIN': 9876,
+    'balance': 100
+}
+
+
+
+for i in range(3):
+    PIN = int(input('PROVIDE PIN: '))
+    try:
+        if len(str(PIN)) !=4:
+            raise Exception
+    except:
+        print('PIN must have 4 digits')
+    else:
+        if PIN != user['PIN']:
+            print('incorrect PIN')
+
+        if PIN == user['PIN']:
+            print('PIN is correct')
+            break
+    finally:
+        print('test run 1')
+else:
+    print('access blocked after 3 attempts!')
+
+print('Amount of withdrawal: \n 1. 20 \n 2. 50 \n 3. 100 \n 4. 200 ')
+
+if PIN == user['PIN']:
+    while True:
+        select = int(input('select one of above: 1, 2, 3, 4: '))
+
+
+        try:
+            if select > 4:
+                raise Exception
+        except:
+                print('ERROR, Select between 1 - 4')
+        else:
+            print('you have:')
+            print(user['balance'])
+            if select ==1:
+                result1 = user['balance'] - 20
+
+                try:
+                    if result1 < 0:
```

Comment:
Here, when you check the result less than 0, you can create a custom error i.e. InsufficientFundError.
Every time the balance fall under 0, you can raise it.
Example:
```python
result = balance - amount
if result <= 0
   raise InsufficientBalanceError
```